src/include/pixels/pixel.hpp

```diff
@@ -1,11 +1,11 @@
 #pragma once
 
-class Filter;
+class Operation;
```

Comment:
I prefer to include operation.hpp more than the forward declaration.

Reply:
I don't want to include it because by design we are going to use the visitor design pattern. If we keep classes forward included in headers where possible (and include them in the .cpp files when needed), we are gonna escape circular dependencies. Also compilation we'll be faster (though our project is small I still think it is good to consider such problems).